src/metadata_validator.py

```diff
@@ -290,14 +290,14 @@ def get_parent_node_cache(self, data_record_parent_nodes):
             if parent_type and parent_id_value and parent_id_value is not None:
                 parent_nodes.append({"type": parent_type, "key": parent_id_property, "value": parent_id_value})
         exist_parent_nodes = self.mongo_dao.search_nodes_by_index(parent_nodes, self.submission[ID])
-        parent_node_cache = set()
+        parent_node_cache = []
         for node in exist_parent_nodes:
             if node.get(PROPERTIES):
                 for key, value in node[PROPERTIES].items():
-                    parent_node_cache.add(tuple([node.get("nodeType"), key, value]))
+                    parent_node_cache.append(tuple([node.get("nodeType"), key, value]))
         return parent_node_cache
     
-    def validate_relationship(self, data_record):
+    def validate_relationship(self, data_record, submission_id):
```

Comment:
submission ID should be included inside data_record, right? If so, there is no need to add submission_id parameter.